gradlew

```diff
@@ -1,54 +1,64 @@
 #!/usr/bin/env bash
+# Created by unknown under unknown
+# Refactored by Jacob Hrbek <kreyren@rixotstudio.cz> under original license in 2020
 
-##############################################################################
-##
-##  Gradle start up script for UN*X
-##
-##############################################################################
+: "
+  Gradle start up script for UN*X
+
+  FIXME: Abstract
+"
+
+# Master unset -- Used to ensure that script does not export unwanted env variables
```

Comment:
This "masterUnset" is ill-conceived. Variables are not exported unless the "export" command is used, and even then, only subsequent child processes receive copies of them. Unsetting before exiting belies a misunderstanding of the shell & its process model.

Reply:
Also, it should be an array.